pandas/tests/arrays/test_datetimelike.py

```diff
@@ -679,7 +687,15 @@ def test_strftime(self, period_index):
         arr = PeriodArray(period_index)
 
         result = arr.strftime("%Y")
-        expected = np.array(period_index.strftime("%Y"))
+        expected = np.array([per.strftime("%Y") for per in arr], dtype=object)
+        tm.assert_numpy_array_equal(result, expected)
+
+    def test_strftime_nat(self):
+        # GH 29578
+        arr = PeriodArray(PeriodIndex(["2019-01-01", pd.NaT], dtype="period[D]"))
```

Comment:
This is working fine for `PeriodArray` and `PeriodIndex` on master; added this test since I didn't see an existing one for this behavior.